lease/lessor.go

```diff
@@ -85,7 +100,10 @@ func (le *lessor) Revoke(id uint64) error {
 
 	delete(le.leaseMap, l.id)
 
-	// TODO: remove attached items
+	for item := range l.itemSet {
+		le.KV.DeleteRange([]byte(item.key), []byte(item.endRange))
```

Comment:
you can lease on a range or on a key.  When a lease expires, the lessor will remove the leased items from the KV.

I think this is good enough. And your concern will complicate the issue a lot. We will not do per-key ownership.